pkg/controllers/nodeclaim/disruption/drift.go

```diff
@@ -81,6 +84,21 @@ func (d *Drift) isDrifted(ctx context.Context, nodePool *v1.NodePool, nodeClaim
 	}); reason != "" {
 		return reason, nil
 	}
+	// Mark the NodeClaim as drifted due to a stale instance type if
+	// 1. The NodeClaim doesn't have the instance type label
+	// 2. The NodeClaim has an instance type that doesn't exist in the cloudprovider instance types
+	// 3. There are no offerings that match the requirements
+	its, err := d.cloudProvider.GetInstanceTypes(ctx, nodePool)
```

Comment:
nit Any thought on moving this check to the separate function like the other drift checks?

Reply:
Would probably be a bit cleaner, will do